### ISSUE TITLE
ButtonSection: avoid click event on non-interactive div

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#729]
For accessibility, attaching `click` events on non-natively interactive elements is far from ideal. HTML provides native interactive elements, they should be used for interaction.

In #714 a click event was added to the container that wraps the button to open the Snippet Preview modal (see `ButtonSection`). It was done to solve an issue with Firefox not opening the modal. See https://github.com/Yoast/wordpress-seo/pull/10685, https://github.com/Yoast/wordpress-seo/issues/10680, and https://github.com/Yoast/wordpress-seo/issues/10682

The real reason of the issue was that the click event was attached to a `<span>` element within the button. A proper fix would have been just attaching the click event on the most proper element: the `<button>`, and not its container.

To my understanding, this would be as simple as passing the `onClick` prop to the Heading created by `wrapInHeading()`, which in turn passes the props to the wrapped component (the button).

See also https://github.com/Yoast/yoast-components/issues/727
